packages/php-wasm/compile/build-assets/phpwasm-emscripten-library.js

```diff
@@ -165,6 +165,92 @@ const LibraryExample = {
 		}
 	},
 
+	js_create_input_device: function (procopenCallId) {
+		if (!PHPWASM.callback_pipes) {
+			PHPWASM.callback_pipes = {};
+		}
+		let dataBuffer = [];
+		let dataCallback;
+		const filename = "proc_id_" + procopenCallId;
+		   const device = FS.createDevice("/dev", filename, function () {
+		}, function (byte) {
+			try {
+				dataBuffer.push(byte);
+				if (dataCallback) {
+					dataCallback(new Uint8Array(dataBuffer));
```

Comment:
if this is only in a Node environment then I think we can `Buffer.concat` them together in `dataBuffer` instead of creating the array, but I don't know if this is important.